tools/gulp/tasks/payload.ts

```diff
@@ -138,16 +142,25 @@ async function uploadPayloadResults(database: firebaseAdmin.database.Database, c
   }
 }
 
-/** Gets the last payload uploaded from previous Travis builds. */
-async function getLastPayloadResults(database: firebaseAdmin.database.Database) {
-  const snapshot = await database.ref('payloads')
-    .orderByChild('timestamp')
-    .limitToLast(1)
-    .once('value');
+/** Gets payload results of the specified commit sha. */
+async function getPayloadResults(database: firebaseAdmin.database.Database, commitSha: string) {
+  const snapshot = await database.ref('payloads').child(commitSha).once('value');
+
+  if (!snapshot.exists()) {
+    throw `There is no payload data uploaded for SHA ${commitSha}`;
+  }
 
-  // The value of the DataSnapshot is an object with the SHA as a key. Later only the
-  // first value of the object will be returned because the SHA is unnecessary.
-  const results = snapshot.val();
+  return snapshot.val();
+}
 
-  return snapshot.hasChildren() ? results[Object.keys(results)[0]] : null;
+/** Gets the SHA of the commit where the payload was uploaded before this Travis Job started. */
+function getCommitFromPreviousPayloadUpload(): string {
+  if (isTravisMasterBuild()) {
+    const commitRange = process.env['TRAVIS_COMMIT_RANGE'];
+    const commitCount = spawnSync('git', ['rev-list', '--count', commitRange]).stdout
```

Comment:
Can you add comments that write out the commit commands and describe what you're getting from them?